pyocd/probe/pydapaccess/interface/pyusb_backend.py

```diff
@@ -282,6 +282,13 @@ def __call__(self, dev):
         except (IndexError, NotImplementedError) as error:
             LOG.debug("Error accessing USB device (VID=%04x PID=%04x): %s", dev.idVendor, dev.idProduct, error)
             return False
+        except ValueError as error:
+            # PyUSB throws "The device has no langid" when it tries to access USB
+            # device in bad state.
+            LOG.debug("Error accessing USB device: %s", error)
+            if "langid" in str(error):
```

Comment:
recognizing the error with regex - very questionable approach.. I would just ignore such a device altogether

Reply:
It throws generic ValueError, ignoring all ValueErrors might catch other errors also. Maybe this is ok. It will leave log row at least.

Reply:
The exception originator function can throw at least 2 types of ValueErrors, https://github.com/pyusb/pyusb/blob/2a434a7c722af65f3c81af2e70721669ed79c284/usb/util.py#L272

Reply:
So, based on https://github.com/pyusb/pyusb/blob/2a434a7c722af65f3c81af2e70721669ed79c284/usb/util.py#L290, does it mean pyusb or pyocd or whatever should not request strings?

Reply:
Seems like a clear device firmware bug https://www.beyondlogic.org/usbnutshell/usb5.shtml.

> String descriptors provide human readable information and are optional. If they are not used, any string index fields of descriptors must be set to zero indicating there is no string descriptor available.

Reply:
> any string index fields of descriptors must be set to zero

Is it? Did you figure out what device creates the disturbance?

Reply:
If the string index field would be zero pyusb get_string would return None. It is not zero but doesn't provide langid and thus raises exception. I did not yet.